app/controllers/shopify_app/sessions_controller.rb

```diff
@@ -36,6 +35,7 @@ def destroy
 
     def authenticate
       if sanitized_shop_name.present?
+        session['shopify.omniauth_params'] = { shop: sanitized_shop_name }
         fullpage_redirect_to "#{main_app.root_path}auth/shopify?shop=#{sanitized_shop_name}"
```

Comment:
You can remove `shop=#{sanitized_shop_name}` from the url and only keep it in the session; so that there's no possible mistake about which one is the right one to use